client/src/components/PropertiesList.tsx

```diff
@@ -1,78 +1,68 @@
-import Loader from "../components/Loader";
-
-import "react-table/react-table.css";
-import "styles/PropertiesList.css";
-
-import withFixedColumns, { TablePropsColumnFixed } from "react-table-hoc-fixed-columns";
-import "react-table-hoc-fixed-columns/lib/styles.css"; // important: this line must be placed after react-table css import
-
 import { I18n } from "@lingui/core";
+import { Trans } from "@lingui/macro";
 import { withI18n } from "@lingui/react";
-import { t, Trans } from "@lingui/macro";
-import { Link } from "react-router-dom";
-import { defaultLocale, SupportedLocale } from "../i18n-base";
-import Helpers, { longDateOptions } from "../util/helpers";
-import { AddressRecord, HpdComplaintCount } from "./APIDataTypes";
-import { withMachineInStateProps } from "state-machine";
-import { AddressPageRoutes } from "routes";
 import classnames from "classnames";
-import { EventProperties, logAmplitudeEvent } from "../components/Amplitude";
-
-export const isPartOfGroupSale = (saleId: string, addrs: AddressRecord[]) => {
-  const addrsWithMatchingSale = addrs.filter((addr) => addr.lastsaleacrisid === saleId);
-  return addrsWithMatchingSale.length > 1;
-};
+import React from "react";
+import { withMachineInStateProps } from "state-machine";
+import "styles/PropertiesList.css";
+import { defaultLocale, SupportedLocale } from "../i18n-base";
+import Helpers from "../util/helpers";
+import Loader from "./Loader";
+import { PortfolioFilters } from "./PortfolioFilters";
+import { MAX_TABLE_ROWS_PER_PAGE, PortfolioTable } from "./PortfolioTable";
 
-const formatAbatementStartYear = (year: number | null, i18n: I18n) => {
-  const thisYear = new Date().getFullYear();
-  if (!year) return null;
-  return year > thisYear ? i18n._(t`Starts ${year}`) : i18n._(t`Since ${year}`);
-};
+// Pattern for context provider to update context from child components
+// https://stackoverflow.com/a/67710693/7051239
 
-const findMostCommonType = (complaints: HpdComplaintCount[] | null) =>
-  complaints && complaints.length > 0 && complaints[0].type;
+export type FilterNumberRange = [number | undefined, number | undefined] | undefined[];
+export const MINMAX_DEFAULT = [undefined, undefined];
 
-/**
- * By default, the `withFixedColumns` hook retypes the Column components within the table
- * as having data arrays of type `unknown`, so this recasting is necessary to make sure
- * we are tyepchecking the raw data as `AddressRecords`
- */
-const ReactTableFixedColumns = withFixedColumns(ReactTable) as React.ComponentType<
-  Partial<TablePropsColumnFixed<AddressRecord, AddressRecord>>
->;
+type FilterValues = {
+  ownernames: string[];
+  unitsres: FilterNumberRange;
+  zip: string[];
+};
 
-const ArrowIcon = () => (
-  <span className="arrow-icon">
-    ↑<span>↓</span>
-  </span>
-);
+export type IFilterContext = {
+  totalBuildings?: number | undefined;
+  filteredBuildings?: number | undefined;
+  filterSelections: FilterValues & { rsunitslatest?: boolean };
```

Comment:
wondering why this one is outside of the FilterValues type?

Reply:
Since the RS units is just a toggle, we don't have any filter options (unlike landlord where we need the list of possible names for the multiselect), so since the selections and options are otherwise the same structure I just made the FilterValues type to reduce the copy/paste.